src/components/cylc/gscan/index.js

```diff
@@ -9,15 +9,18 @@
 function getWorkflowSummary (workflow) {
   const states = new Map()
   for (const taskProxy of workflow.taskProxies) {
-    for (const job of taskProxy.jobs) {
-      // TODO: temporary fix, as the backend is sending ready jobs, but they will change in cylc flow&uiserver in the future
-      if (job.state === 'ready') {
-        continue
+    // a task in waiting, may not have any jobs
```

Comment:
This appeared after I re-generated the mock data. Interestingly it got some tasks with no jobs, which caused console errors in the UI.

Reply:
Does the task task have jobs set to `null` rather than `[]`?

Reply:
Funny, that's what I thought. That I had seen an `undefined`, not even a `null`. But trying now I only got empty `[]` jobs. Guess this is not necessary. Let me fix it.

Reply:
Ah, actually, re-reading my comment above I remembered it was recorded in the `checkpoint.js`: 

https://github.com/cylc/cylc-ui/blob/8b8c3da7a1c054ae15a7b4f38ccd49957ceaca95/src/services/mock/checkpoint.js#L293-L308

Can't say what caused it. I re-generated it as I normally do, and then got this `undefined` error in the offline mode.

Do you think we should ignore as it shouldn't happen with live data? Maybe something with the way we produce the mock data, or something I could have done wrong @oliver-sanders ?

Reply:
Mmm, I cant remember how we generate checkpoints, will have a dig...